proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,641 @@
+
+## Out-of-scope
+
+* We won't support saving the machine-readable output to disk as part of this spec. The work-around is for the consumer to read from the console's stdout stream.
+* At this point, no other CLI commands (e.g. dotnet list reference) will be within scope for this feature.
+"--parsable" option needs separate spec.
+* Currently license info is not emitted from any cli command, it could be quite useful, we should consider in the future.
+
+## Rationale and alternatives
+Currently, no other `dotnet command` implemented this, this is the 1st time dotnet command implementing `json`(etc) output, so it could become example for others next time they implement.
+Please note, except "tab completion" (for dotnet) part all changes would be inside NuGet.Client repo(under NuGet.Core), and risk of introducing regression is low.(`--format text` refactoring related changes only come into my mind.), no impact on dotnet sdk.
+
+## Prior Art
+
+<!-- What prior art, both good and bad are related to this proposal? -->
+<!-- Do other features exist in other ecosystems and what experience have their community had? -->
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevent to this proposal? -->
+
+* https://github.com/NuGet/Home/blob/dotnet-audit/proposed/2021/DotNetAudit.md#dotnet-audit---futjson There're some overlaps, but current spec is one more focused on SBOM and CI/CD actions, while `dotnet audit fix` is more focused detecting/fixing dependencies manually. Current spec already include ideas from this spec like `json format`.
+
+* https://github.com/NuGet/Home/wiki/%5BSpec%5D-Machine-readable-output-for-dotnet-list-package Basic idea from this spec is still same here and I extended from it. In current spec more orient to `dotnet style syntax` and cover more uses cases like `dotnet list package --vulnerable --format json` and `--include-transitive`, also json schema improved to include project name/identifier for multi-project scenario which would most likely use case.
+
+## Unresolved Questions
+
+* Chris: One problem is most `dotnet list package` options outputs are exclusive and have to query separately and join result to get full picture. Both below approach require additional work.
+  * dotnet cli needs an all up `dotnet list package --all` that shows all deprecated, vulnerable, and outdated top level and transitive packages. [r766860629](https://github.com/NuGet/Home/pull/11446#discussion_r766860629)
+  * Alternatively the behavior of --outdated and --deprecated could be additive rather than exclusive. That adds to the scope of this work though.[r766860629](https://github.com/NuGet/Home/pull/11446#discussion_r766860629). With this approach we can have single schema to populate.
```

Comment:
@nkolev92 
What is preventing `--outdated` and `--deprecated` being additive rather than exclusive? I saw spec is designed this way, but it doesn't really explain why. Is there technical reason?

Reply:
Deprecated and outdated are both filters. 

What does --outdated + --deprecated do?

Packages that are both outdated and deprecated? 
Packages that are either outdated or deprecated? 

The way the oudated output is created today, you don't really get information about the other packages. 
The deprecated output shape is slightly different.

Reply:
Perhaps these questions shouldn't be answered today. They also are questions that I hope https://github.com/NuGet/Home/pull/11549 answers more. I don't think it's very intuitive to have an `--all` for a list command. The list command can list stuff and the audit command can audit stuff.